### PR TITLE
fix: Updating upgrade factories for 8.6 to 8.7

### PR DESCRIPTION
**Pull request by grlimacan**
## Description

fix: Updating upgrade factories for 8.6 to 8.7

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #

### CODE REVIEW

File: optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/Upgrade86to87PlanFactory.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class Upgrade86to87PlanFactory implements UpgradePlanFactory {
+
+  @Override
+  public UpgradePlan createUpgradePlan(final UpgradeExecutionDependencies dependencies) {
+    validateOptimizeModeAndFailIfC7DataPresent(
+        dependencies.databaseClient(), dependencies.indexNameService());
+    return UpgradePlanBuilder.createUpgradePlan().fromVersion("8.6").toVersion("8.7.0").build();
```
Comment (misiekhardcore) [inline optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/Upgrade86to87PlanFactory.java:22]: shouldnt it be full version for both from and to?

```suggestion
    return UpgradePlanBuilder.createUpgradePlan().fromVersion("8.6.0").toVersion("8.7.0").build();
```

**Review by misiekhardcore — APPROVED** [review]
LGTM, just one comment

File: optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/Upgrade86to87PlanFactory.java (hunk 1 above)
Comment (grlimacan) [reply optimize/upgrade/src/main/java/io/camunda/optimize/upgrade/plan/factories/Upgrade86to87PlanFactory.java:22]: I believe it's correct the way it is, that's how all previous factories looked

**Review by misiekhardcore — APPROVED** [review]

**Comment by misiekhardcore** [comment]
@buccarel I think the fix in [Giuliano's PR](https://github.com/camunda/camunda-docs/pull/4398) will do the trick, he'll just take a look at the failing OS integrity test

**Comment by grlimacan** [comment]
> @grlimacan will you be adding the fix for OS to this PR or will do it in a separate one?

I think it makes sense to do a separate one. I will get to it in about 1,5h I estimate. I will rebase this now so that we can get it merged to the release branch. I wonder where this conflict now comes from

**Comment by grlimacan** [comment]
> > @grlimacan will you be adding the fix for OS to this PR or will do it in a separate one?
> 
> I think it makes sense to do a separate one. I will get to it in about 1,5h I estimate. I will rebase this now so that we can get it merged to the release branch. I wonder where this conflict now comes from

@misiekhardcore I rebased it, once pipelines are green (except for the one we know is problematic) feel free to merge to release branch. I will create a new PR from the then merged release branch to do the OpenSearch fix caused by the client update

**Pull request merged**